### PR TITLE
add istio1.14 to pixiu

### PR DESCRIPTION
**Pull request by bobtthp**
## Modification
1. Merge istio version to istio 1.14.3
2. Golang dependency version is modified to 1.18
3. The package in the project is replaced by istio.io/istio -> github.com/apache/dubbo-go-pixiu
4. Istio package name conflict replace pkg/config -> pkg/pixiu_config pkg/cluster -> pkg/pixiu_cluster
5. go.etcd.io/etcd/server and pixiu dependency reference go.opentelemetry.io/otel/semconv Upgrade etcd/server version v3.6.0-alpha.0
6. Fix istio code to comply with golint-ci requirements
7. Fix istio code to pass all tests (6a90733946e6524902246b6ba24672de5d6c6f41)
8. Improve the description of the NOTICE file and the version dependency information of the LICENSE folder
9. Since the config pkg of pixiu has been renamed to pixiu-config pkg, the import needs to be replaced on the samples side. Now, the sed replacement in the script has been run first (c30fc5300b3d9916d3d51e47d9d7089864db72d2)

## 修改点：
1. 合并istio版本为 istio 1.14.3
2. golang 依赖版本修改为 1.18
3. 项目内package替换 istio.io/istio -> github.com/apache/dubbo-go-pixiu
4. istio 包名冲突替换 pkg/config -> pkg/pixiu_config pkg/cluster -> pkg/pixiu_cluster
5. go.etcd.io/etcd/server 和 pixiu 依赖引用 go.opentelemetry.io/otel/semconv 升级etcd/server 版本 v3.6.0-alpha.0
6. 修复 istio 代码符合 golint-ci 的要求
7. 修复 istio 代码通过所有测试 (6a90733946e6524902246b6ba24672de5d6c6f41)
8. 完善 NOTICE 文件描述、完善 LICENSE 文件夹的版本依赖信息
9. 由于 pixiu 的 config pkg 重命名为 pixiu-config pkg 了，需要在 samples 那边换下 import，现在是在脚本里面 sed 替换先跑过了 (c30fc5300b3d9916d3d51e47d9d7089864db72d2)

### CODE REVIEW

File: istioctl/cmd/add-to-mesh_test.go [hunk 1]
```diff
@@ -0,0 +1,395 @@
...
+	"strings"
+	"testing"
+
+	appsv1 "k8s.io/api/apps/v1"
+	coreV1 "k8s.io/api/core/v1"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/dynamic/fake"
```
Comment (baerwang) [inline istioctl/cmd/add-to-mesh_test.go:30]: use import tool format

Comment (bobtthp) [reply istioctl/cmd/add-to-mesh_test.go:30]: get

**Comment by thehackercat** [comment]
Should we use a git sub folder or a separated folder like `pkg/istio` to store those istio control pane codes? Which makes it easily to maintain codes if we are going to upgrade istio version or refractor, we can make changes in specified folder rather than scattered files.

**Comment by pinxiong** [comment]
> Should we use a git sub folder or a separated folder like `pkg/istio` to store those istio control pane codes? Which makes it easily to maintain codes if we are going to upgrade istio version or refractor, we can make changes in specified folder rather than scattered files.

👍 I totally aggre with you

**Comment by pinxiong** [comment]
This is a huge issue that makes it difficult to review. I think we can split it into several different issues.

**Comment by chickenlj** [comment]
> Should we use a git sub folder or a separated folder like `pkg/istio` to store those istio control pane codes? Which makes it easily to maintain codes if we are going to upgrade istio version or refractor, we can make changes in specified folder rather than scattered files.

Merging Istio into Pixiu is gonna be a one-shot thing, we are not planning to follow Istio with the latest versions in the future.

Even though, I think it's still worth discussing whether we should place code from Istio into one single parent folder to make it more clear to maintain at first. And then after that, we can gradually move codes out of Istio or add codes of our own that extend Istio in our own pkg.

**Review by MasterKenway — APPROVED** [review]

**Review by mark4z — APPROVED** [review]
LGTM

**Review by AlbumenJ — APPROVED** [review]
LGTM

**Pull request merged**